### DISCUSSION TITLE
Publish older versions of the Eclipse plugin?

### DISCUSSION THREAD

**Discussion by DominicAnelloSonos** [spotbugs/spotbugs#1636] [General]
Are there older versions of the eclipse plugin available?  It appears the repository at `https://spotbugs.github.io/eclipse/` only has the latest build available.  I recently tried to roll back my eclipse workspace configuration and received this error:

```
No repository found containing: org.eclipse.update.feature,com.github.spotbugs.plugin.eclipse,4.2.3.r202104160407-de5da7e
```

We are not ready to update to `4.3.0` yet and I'd like my Eclipse configuration to match what we have configured in our CI build environment/Maven.  I checked the releases in GitHub but the plugin build wasn't published there.  

Even if I didn't want to revert SpotBugs version but wanted to roll back my Eclipse workspace for some other reason,  this would be an issue since Eclipse doesn't let you pick and choose what plugins you revert and if one fails the entire rollback fails.  So having the SpotBugs plugin installed essentially disables this Eclipse feature.

**Comment by KengoTODA** [comment]
Previously we uploaded a plugin zip to each Release, like https://github.com/spotbugs/spotbugs/releases/tag/4.0.3

so it's a bug of current release procedure, we need to upload not only `spotbugs/build/distributions/*` but also `eclipsePlugin/build/distributions/eclipsePlugin.zip`.

https://github.com/spotbugs/spotbugs/blob/9868ac4114ed054fe6c53dfcffda49d785cf2186/.github/workflows/release.yml#L80

I'll fix this problem for future release like `4.3.1`. But you need to build eclipse plugin by your own for version `4.0.4`~`4.3.0`.

**Reply by KengoTODA** [reply]
issued #1637

**Reply by KengoTODA** [reply]
I've confirmed that 4.4.0 release has a plugin attached to the Release page: https://github.com/spotbugs/spotbugs/releases/tag/4.4.0

**Reply by DominicAnelloSonos** [reply]
Thank you!

**Comment by DominicAnelloSonos** [comment]
Thanks @KengoTODA, I appreciate it. 

Also, I tried to build the 4.2.3 tag, but I think there is some misconfiguration in the gralde build for JDK 8, which is what the [eclipse instructions](https://github.com/spotbugs/spotbugs/blob/release-3.1/eclipsePlugin/doc/building_spotbugs_plugin.txt) say to use.  The error is an unrecognized flag to `javac` called `--release`,  which I think was introduced in Java 9.

```
C:\src\spotbugs>.\gradlew eclipse

> Configure project :spotbugs
The signing key and password are null. This can be ignored if this is a pull request.

> Configure project :spotbugs-annotations
The signing key and password are null. This can be ignored if this is a pull request.

> Configure project :spotbugs-ant
The signing key and password are null. This can be ignored if this is a pull request.

> Configure project :test-harness
The signing key and password are null. This can be ignored if this is a pull request.

> Configure project :test-harness-core
The signing key and password are null. This can be ignored if this is a pull request.

> Configure project :test-harness-jupiter
The signing key and password are null. This can be ignored if this is a pull request.

> Task :spotbugs-annotations:compileJava FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':spotbugs-annotations:compileJava'.
> invalid flag: --release

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 8s
8 actionable tasks: 7 executed, 1 up-to-date

C:\src\spotbugs>C:\bin\jdk-1.8.0_181\bin\java -version
java version "1.8.0_181"
Java(TM) SE Runtime Environment (build 1.8.0_181-b13)
Java HotSpot(TM) 64-Bit Server VM (build 25.181-b13, mixed mode)

C:\src\spotbugs>echo %JAVA_HOME%
C:\bin\jdk-1.8.0_181
```

**Reply by KengoTODA** [reply]
4.2.3 is expected to built with JDK 11 or later. Our release build uses OpenJDK 11: https://github.com/spotbugs/spotbugs/blob/4.2.3/.github/workflows/release.yml#L14-L17

**Reply by DominicAnelloSonos** [reply]
Thanks!  I eventually figured out that I needed JVM 11.

I ended up having to spin up a linux VM because it was producing non-functional builds on Windows and MacOS: it would build an `eclipsePlugin.zip`, but would not run when installed.  No errors in the eclipse log, just no spotbugs menu item or views.  However, after building under linux it seems to be working.  Thanks so much again!

**Comment by hazendaz** [comment]
closing as resolved.
